src/db.c

```diff
@@ -287,14 +287,11 @@ static void dbSetValue(redisDb *db, robj *key, robj *val, int overwrite, dictEnt
         /* Because of RM_StringDMA, old may be changed, so we need get old again */
         old = dictGetVal(de);
     }
-    dict *d = kvstoreGetDict(db->keys, slot);
-    dictSetVal(d, de, val);
-
+    kvstoreDictSetVal(db->keys, slot, de, val);
     if (server.lazyfree_lazy_server_del) {
         freeObjAsync(key,old,db->id);
     } else {
-        /* This is just decrRefCount(old); */
-        d->type->valDestructor(d, old);
```

Comment:
i was wondering why it was like that.
i suppose it was an attempt to release a dictEntry value, without assuming anything about which destructor this value uses.
i see it was added in c84248b5d2 in place of `dictFreeVal(db->dict, &auxentry)`
i suppose it's not that bad for db.c to assume it knows how values are destructed, but wanted to raise that for discussion.
@zuiderkwast FYI.

Reply:
I think I wrote it for some reason of abstraction. Don't remember why. Calling decrRefCound directly is totally fine.